examples/graphs/node2vec_movielens.py

```diff
@@ -0,0 +1,584 @@
+rated_movies = ratings[ratings.rating >= min_rating]
+# Group instances by user.
+movies_grouped_by_users = list(rated_movies.groupby("userId"))
+for group in tqdm(
+    movies_grouped_by_users,
+    position=0,
+    leave=True,
+    desc="Compute movie rating frequencies",
+):
+    # Get a list of movies rated by the user.
+    current_movies = list(group[1]["movieId"])
+
+    for i in range(len(current_movies)):
+        item_frequency[current_movies[i]] += 1
+        for j in range(i + 1, len(current_movies)):
+            x = min(current_movies[i], current_movies[j])
+            y = max(current_movies[i], current_movies[j])
+            pair_frequency[(x, y)] += 1
+
+"""
+### Step 2: create the graph with the nodes and the edges
+
+To reduce the number of edges between nodes, we only add an edge between movies
+if the weight of the edge is greater than `min_weight`.
+"""
+
+min_weight = 10
+D = math.log(sum(item_frequency.values()))
+
+# Create the movies undirected graph.
+movies_graph = nx.Graph()
+# Add weighted edges between movies.
+# This automatically adds the movie nodes to the graph.
+for pair in tqdm(
+    pair_frequency, position=0, leave=True, desc="Creating the movie graph"
+):
+    x, y = pair
+    xy_frequency = pair_frequency[pair]
+    x_frequency = item_frequency[x]
+    y_frequency = item_frequency[y]
+    pmi = math.log(xy_frequency) - math.log(x_frequency) - math.log(y_frequency) + D
+    weight = pmi * xy_frequency
+    # Only include edges with weight >= min_weight.
+    if weight >= min_weight:
+        movies_graph.add_edge(x, y, weight=weight)
+
+"""
+Let's display the total number of nodes and edges in the graph.
+Note that the number of nodes is less than the total number of movies,
+since only the movies that have edges to other movies are added.
+"""
+
+print("Total number of graph nodes:", movies_graph.number_of_nodes())
+print("Total number of graph edges:", movies_graph.number_of_edges())
+
+"""
+Let's display the average node degree (number of neighbours) in the graph.
+"""
+
+degrees = []
+for node in movies_graph.nodes:
+    degrees.append(movies_graph.degree[node])
+
+print("Average node degree:", round(sum(degrees) / len(degrees), 2))
+
+"""
+### Step 3: Create vocabulary and a mapping from tokens to integer indices
+
+The vocabulary is the nodes (movie ids) in the graph.
+"""
+
+vocabulary = ["NA"] + list(movies_graph.nodes)
+vocabulary_lookup = {token: idx for idx, token in enumerate(vocabulary)}
+
+"""
+## Implement Biased Random Walk
+
+A random walk starts from a given node, and randomly picks a neighbour node to move to.
+If the edges are weighted, the neighbour is selected *probabilistically* with
+respect to weights of the edges between the current node and its neighbours.
+This procedure is repeated for `num_steps` to generate a sequence of *related* nodes.
+
+The *biased* random walk balances between **breadth-first sampling**
```

Comment:
Suggestion:

- Add a link for more info on https://en.wikipedia.org/wiki/Biased_random_walk_on_a_graph to help new users
 
```suggestion
The [*biased* random walk](https://en.wikipedia.org/wiki/Biased_random_walk_on_a_graph) balances between **breadth-first sampling**
```